src/stores/CheckoutController.ts

```diff
@@ -172,3 +179,43 @@ export const paypalPayment = async (successUrl: string, cancelUrl: string, error
     });
   });
 };
+
+export const getSubscriptionSwitches = async (): Promise<unknown> => {
+  const { cleengSandbox } = useConfigStore.getState().getCleengData();
+  const { user, subscription, auth } = useAccountStore.getState();
+
+  if (!user || !subscription || !auth) return;
+  const response = await checkoutService.getSubscriptionSwitches(
+    {
+      customerId: user.id,
+      offerId: subscription.offerId,
+    },
+    cleengSandbox,
+    auth.jwt,
+  );
+
+  if (!response.responseData.available.length) return;
+
+  const switchOffers = response.responseData.available.map((offer: SwitchOffer) => checkoutService.getOffer({ offerId: offer.toOfferId }, cleengSandbox));
+  const offers = await Promise.all(switchOffers);
+
+  // Sort offers for proper ordering in "Choose Offer" modal when applicable
+  const offerSwitches = offers.sort((a, b) => a?.responseData.offerPrice - b?.responseData.offerPrice).map((item) => item.responseData);
+  useCheckoutStore.setState({ offerSwitches });
+};
+
+export const switchSubscription = async (toOfferId: string, switchDirection: 'upgrade' | 'downgrade') => {
+  const { cleengSandbox } = useConfigStore.getState().getCleengData();
+  const { user, subscription, auth } = useAccountStore.getState();
+  if (!user || !subscription || !auth) return;
+
+  const payload = { toOfferId, customerId: user.id, offerId: subscription.offerId, switchDirection: switchDirection, jwt: auth.jwt };
```

Comment:
This method does need to throw when the method is not defined in the `checkoutService`:

```suggestion
  if (!('switchSubscription' in checkoutService)) throw new Error('SwitchSubscription not supported');
  
  const payload = { toOfferId, customerId: user.id, offerId: subscription.offerId, switchDirection: switchDirection, jwt: auth.jwt };
```